src/dependency_injector/providers.pyx

```diff
@@ -204,7 +204,11 @@ cdef class Provider(object):
         if self.__last_overriding is not None:
             result = self.__last_overriding(*args, **kwargs)
         else:
-            result = self._provide(args, kwargs)
+            try:
+                result = self._provide(args, kwargs)
+            except Exception as exc:
+                print(self)
```

Comment:
I'm not quite sure about using `print()` here. It will send some data to stdout, which might not always be the desired behavior. 

The idea, generally, looks interesting. Maybe we can use the ``logging`` module?

Reply:
I would agree with @rmk135 to use the `logging` module with `logger.exception` with some extra useful messages as it allows users to have more control on the output through `logging.config`. Meanwhile, I believe the [LogRecord](https://docs.python.org/3/library/logging.html#logging.LogRecord) may not provide precise info given the cython context (caller line number, stacktrace etc.)

Reply:
Thank you for reply!

@rmk135 @gen-xu 

Please look at updated pull request.
I hope logging.debug() is good idea because, by default, we have warning log level, and I hope no one use debug level in production) - so nothing will change for current users until logging level set to debug.
If this request will be accepted - I'll add article to documentation about this.